src/main/java/me/moodcat/backend/rooms/RoomInstance.java

```diff
@@ -58,6 +62,11 @@
     private final UnitOfWorkSchedulingService unitOfWorkSchedulingService;
 
     /**
+     * The profanity checker to filter out 'bad' chatmessages.
+     */
+    private final ProfanityChecker profanityChecker = new ProfanityChecker();
```

Comment:
Setting this through the constructor and instantiating through Guice will technically make this instance singleton, which saves memory. It also makes the implementation more testable/mockable. So I agree with Tim on this.